source/opt/module.cpp

```diff
@@ -117,5 +119,22 @@ void Module::ToBinary(std::vector<uint32_t>* binary, bool skip_nop) const {
   ForEachInst(write_inst, true);
 }
 
+uint32_t Module::ComputeIdBound() const {
+  uint32_t highest = 0;
+
+  ForEachInst(
+      [&highest](const Instruction* inst) {
+        // Use a const-cast just to access begin() and end() for the range-for.
+        for (const auto& operand : *const_cast<Instruction*>(inst)) {
```

Comment:
Sorry for this. It seems we can get away with this cast by adding the following into `Instruction`?

```cpp
const_iterator begin() const { return operands_.cbegin(); }
const_iterator end() const { return operands_.cend(); }

```